From a5d625fe194f9d02a02dd9995276b0aa7a57a1f8 Mon Sep 17 00:00:00 2001
From: alfonsonishikawa <alfonsonishikawa@unknown>
Date: Tue, 2 Apr 2013 21:14:44 +0000
Subject: [PATCH] GORA-174: Updated
 DataStoreTestUtil#testGetEmployeeWithFields() to test only on column subsets
 with values.

git-svn-id: https://svn.apache.org/repos/asf/gora/branches/GORA_174@1463755 13f79535-47bb-0310-9956-ffa450edef68
---
 .../java/org/apache/gora/store/DataStoreTestUtil.java | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/gora-core/src/test/java/org/apache/gora/store/DataStoreTestUtil.java b/gora-core/src/test/java/org/apache/gora/store/DataStoreTestUtil.java
index aad3cdb0f..4a959537b 100644
--- a/gora-core/src/test/java/org/apache/gora/store/DataStoreTestUtil.java
+++ b/gora-core/src/test/java/org/apache/gora/store/DataStoreTestUtil.java
@@ -244,7 +244,14 @@ public static void testGetEmployeeWithFields(DataStore<String, Employee> dataSto
     dataStore.put(ssn, employee);
     dataStore.flush();
 
-    String[] fields = employee.getFields();
+    // XXX See GORA-216: special case until later reviewed.
+    // Like in K-V stores, if retrieved column does not exists ([webpage] case),
+    // get() must return 'null'.
+    // We prepare an actual weird synthetic test.
+    
+    // String[] fields = employee.getFields();
+    String[] fields = {"name","dateOfBirth","ssn","salary"} ;
+    
     for(Set<String> subset : StringUtils.powerset(fields)) {
       if(subset.isEmpty())
         continue;
@@ -255,7 +262,7 @@ public static void testGetEmployeeWithFields(DataStore<String, Employee> dataSto
         expected.put(index, employee.get(index));
       }
 
-      Assert.assertEquals(expected, after);
+      Assert.assertEquals(expected, after);        
     }
   }
 
